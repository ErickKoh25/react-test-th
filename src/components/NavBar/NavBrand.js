import React from 'react'

export const NavBrand = ({img}) => {
    return (
        <div className='navbar-brand'>
            
            {
                img ? 
                    <img src={img} alt='logo-aerolinea'/>
                :   <h2>
                        Mi Aerolinea
                        <i class="fas fa-plane"></i>
                    </h2>
            }

            
        </div>
    )
}
