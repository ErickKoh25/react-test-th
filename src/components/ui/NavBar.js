import React from 'react'

export const NavBar = () => {
    return (
        <nav className='navbar'>
            <div className='navbar-brand'>
                {/* <img/> */}
                <h2>Mi Aerolinea</h2>
            </div>
            <div class='menu'>
                <ul>
                    <li>
                        <a>
                            Mis Reservaciones
                        </a>
                    </li>
                    <li>
                        <a>
                            Registrate
                        </a>
                    </li>
                    <li>
                        <a>
                            Iniciar SesiÃ³n
                        </a>
                    </li>                        
                </ul>
            </div>
        </nav>
    )
}
