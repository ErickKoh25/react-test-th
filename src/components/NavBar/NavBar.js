import React, { memo } from 'react'
import { NavBrand } from './NavBrand/NavBrand';
import { NavMenu } from './NavMenu/NavMenu';
export const NavBar = memo(() => {
    return (
        <nav className='navbar'>
            <NavBrand/>
            <NavMenu />
        </nav>
    )
})
