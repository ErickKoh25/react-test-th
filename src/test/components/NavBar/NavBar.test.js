import '../../../setupTests';
import React from 'react'
import {mount} from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'
import { NavBar } from '../../../components/NavBar/NavBar';
import { MemoryRouter } from 'react-router-dom'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}

let store = mockStore(initState)

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <NavBar/> 
        </MemoryRouter>
    </Provider>
)

describe('Pruebas en NavBar', () => {
    
    test('Debe mostrarse correctamente', ()=>{
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.navbar').exists()).toBe(true)
    })
})