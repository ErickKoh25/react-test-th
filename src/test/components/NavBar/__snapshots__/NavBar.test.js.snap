// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en NavBar Debe mostrarse correctamente 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-brand"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h2>
        Mi Aerolinea
        <i
          className="fas fa-plane"
        />
      </h2>
    </a>
  </div>
  Array [
    <i
      className="fas fa-bars pointer"
      onClick={[Function]}
    />,
    <div
      className="menu "
    >
      <ul>
        <li
          key="0"
        >
          <a
            className=""
            href="/my-reservations"
            onClick={[Function]}
          >
             
            Mis Reservaciones
          </a>
        </li>
      </ul>
    </div>,
  ]
</nav>
`;
